test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,263 @@
+              test,
+              403,
+              'You do not have permission to perform the requested action',
+              'verify-custom-factor custom-factor-form',
+              {
+                status: 403,
+                responseType: 'json',
+                responseText: '{"errorCode":"E0000006","errorSummary":"You do not have permission to perform the requested action","errorLink":"E0000006","errorId":"oae3CaVvE33SqKyymZRyUWE7Q","errorCauses":[]}',
+                responseJSON: {
+                  errorCode: 'E0000006',
+                  errorSummary: 'You do not have permission to perform the requested action',
+                  errorLink: 'E0000006',
+                  errorId: 'oae3CaVvE33SqKyymZRyUWE7Q',
+                  errorCauses: [],
+                },
+              }
+            );
+          });
+      });
+    });
+
+    Expect.describe('when SKIP_IDP_FACTOR_VERIFICATION_BUTTON FF is off', function() {
+      itp('shows the right subtitle', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': false } ).then(function(test) {
+          expectSubtitleToBe(test, 'Clicking below will redirect to verification with IDP factor');
+        });
+      });
+
+      itp('shows the button bar', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': false } ).then(function(test) {
+          expect(test.form.buttonBar().hasClass('hide')).toBe(false);
+        });
+      });
+
+      itp('hides the waiting spinner', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': false } ).then(function(test) {
+          expect(test.form.hasSpinner().length).toBe(0);
+        });
+      });
+
+      itp('redirects to third party when Verify button is clicked', function() {
+        spyOn(SharedUtil, 'redirect');
+        return setupClaimsProviderFactorWithIntrospect()
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
+            test.form.submit();
+            return Expect.waitForSpyCall(SharedUtil.redirect);
+          })
+          .then(function() {
+            expect(SharedUtil.redirect).toHaveBeenCalledWith(
+              'http://rain.okta1.com:1802/sso/idps/idpId?stateToken=testStateToken'
+            );
+          });
+      });
+
+      itp('does not redirect to third party automatically', function() {
+        spyOn(SharedUtil, 'redirect');
+        return setupClaimsProviderFactorWithIntrospect({ 'features.skipIdpFactorVerificationBtn': false }, null)
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
+            test.form.initialize();
+            expect(SharedUtil.redirect).not.toHaveBeenCalledWith(
+              'http://rain.okta1.com:1802/sso/idps/idpId?stateToken=testStateToken'
+            );
```

Comment:
We don't expect a redirect to occur at all, right? If that is the case, let's change this to:

```suggestion
            expect(SharedUtil.redirect).not.toHaveBeenCalled();
```

or

```suggestion
            expect(SharedUtil.redirect).toHaveBeenCalledTimes(0);
```